src/depends/common/FixedHash.h

```diff
@@ -338,19 +339,11 @@ namespace dev
         void clear() { ref().cleanse(); }
     };
 
-    /// Fast equality operator for h256.
-    template<> inline bool FixedHash<32>::operator==(FixedHash<32> const& _other) const
-    {
-        const uint64_t* hash1 = (const uint64_t*)data();
-        const uint64_t* hash2 = (const uint64_t*)_other.data();
-        return (hash1[0] == hash2[0]) && (hash1[1] == hash2[1]) && (hash1[2] == hash2[2]) && (hash1[3] == hash2[3]);
-    }
-
     /// Fast std::hash compatible hash function object for h256.
     template<> inline size_t FixedHash<32>::hash::operator()(FixedHash<32> const& value) const
     {
-        uint64_t const* data = reinterpret_cast<uint64_t const*>(value.data());
-        return boost::hash_range(data, data + 4);
+        auto data = std::bit_cast<std::array<uint64_t, 4>>(value.m_data);
+        return boost::hash_range(data.begin(), data.end());
```

Comment:
Why are we hashing a hash? You're not going to make it any more random, you know! Perhaps worth a ticket to investigate deleting?

Reply:
My guess is that whoever wrote this tried to make it more "efficient" somehow by hashing the uint64_t instead of the actual characters.
It's possible to delete this specialization as well, but I thought it might be a bit safer to do the same thing in case there's some code that might break if this is changed.
So this will produce the same hash as it always has. After the release we can probably investigate if it's worth changing.